.github/workflows/common.yml

```diff
@@ -3,7 +3,11 @@ on:
   push:
     branches:
       - master
+    paths:
```

Comment:
Consider adding `paths-ignore` to exclude files that don't need testing, such as:
```yaml
paths-ignore:
  - '**.md'
  - 'docs/**'
  - '.gitignore'
```
This can further optimize CI runs.